src/engine/builtins.c

```diff
@@ -2515,15 +2506,16 @@ LIST * builtin_shell( FRAME * frame, int flags )
         buffer[ ret ] = 0;
         if ( !no_output_opt )
         {
-            if ( strip_eol_opt )
-                rtrim( buffer );
```

Comment:
Previously, this would occur _inside_ the loop which is split on 1024 byte boundaries in commands.